extractor/src/main/java/org/schabi/newpipe/extractor/search/filter/SearchFiltersBase.java

```diff
@@ -0,0 +1,138 @@
+package org.schabi.newpipe.extractor.search.filter;
+
+import org.schabi.newpipe.extractor.services.youtube.search.filter.YoutubeFilters;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Deriving class must always call {@link SearchFiltersBase#init()} and {@link SearchFiltersBase#build()}
+ * in constructor manually.
+ */
```

Comment:
This is what I wanted to do. But as in one derived class I rely on other objects that are  not yet initialized because the derived class is not yet constructed. I'll look into it again